framework/base/src/types/managed/wrapped/fixed_point_biguint.rs

```diff
@@ -0,0 +1,101 @@
+use crate::{api::ManagedTypeApi, types::BigUint};
+
+use core::ops::{Add, Div, Mul, Sub};
+
+#[derive(Debug, Clone)]
+pub struct FixedPoint<M: ManagedTypeApi, const DECIMALS: usize> {
+    data: BigUint<M>,
+}
+
+impl<M: ManagedTypeApi, const DECIMALS: usize> FixedPoint<M, DECIMALS> {
+    pub fn scaling_factor() -> BigUint<M> {
+        // TODO: cache
+        BigUint::from(10u32).pow(DECIMALS as u32)
+    }
+
+    pub fn trunc(&self) -> BigUint<M> {
+        &self.data / &Self::scaling_factor()
+    }
+
+    pub fn data(&self) -> &BigUint<M> {
```

Comment:
The name is not great.
More like "into_raw_units".
We also need "from_raw_units".